w6_d2_2_ruby_methods/README.md

```diff
@@ -0,0 +1,284 @@
+# Intro To Ruby: Methods
+
+
+| Objectives: Students will be able to... |
+|:--- |
+| Compare and contrast functions in Ruby and JavaScript. |
+| Write methods in Ruby to solve problems. |
+
+### Parts of A Language
+
+* `Primitives`
+* `Combinations`
+* **`Abstractions`**
+
+
+### Some Abstractions
+
+#### Javascript
+
+* Function
+    * Anonymous: `function (param1, [..param2, [...]]){...}`,
+    * Named: `function Name(param1, [..param2, [...]]){...}`
+    * Use *lexical scope* (look up variables in outer scopes)
+    * Captures scope (think of closures)
+    * Used as values, passed around like other variables
+    * Require explicit `return`
+    * All parameters are optional
+
+#### Ruby
+
+ * Function
+    * Uses `def`
+    * Does not capture scope
+    * Not used as values or passed around
+    * Optional parameters must be specified
+    * Implicitly returns last evaluation
+
+* Block
+    * Used with `.each`, `.map`, et cetera
+
+    ```
+    some_method do |param1, [param2, [...]]
+        # some code
+    end
+
+    ```
+    * Captures scope
+
+##Ruby Methods
+
+### Define a method
+
+```ruby
+def say_hello
+  puts "Hello"
+end
+
+say_hello
+```
+
+### Define a method with a parameter
+
+```ruby
+def say(something)
+  puts something
+end
+
```

Comment:
Should we add a comment here like: # In Ruby you can call a method that takes parameters with or without parentheses